test_rclcpp/test/test_multithreaded.cpp

```diff
@@ -50,7 +50,7 @@ static inline void multi_consumer_pub_sub_test(bool intra_process)
     [&counter, &intra_process](test_rclcpp::msg::UInt32::ConstSharedPtr msg,
       const rmw_message_info_t & info) -> void
     {
-      ++counter;
+      counter.fetch_add(1);
```

Comment:
as a side note, I think `fetch_add(1)` and `++counter` are actually equivalent (if they aren't, it's a bug in gcc). Using `fetch_add` makes it more obvious to the reader than the variable is atomic, however. (Same with `store` down below.)

http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith
http://en.cppreference.com/w/cpp/atomic/atomic/operator%3D